src/EnergyPlus/MixedAir.cc

```diff
@@ -764,12 +764,10 @@ namespace MixedAir {
                                                                              sensOut,
                                                                              latOut);
                 }
-                if (AirLoopInputsFilled) UnitarySystems::UnitarySys::getUnitarySysHeatCoolCoil(CompName, OACoolingCoil, OAHeatingCoil, 0);
                 if (MyOneTimeCheckUnitarySysFlag(OASysNum)) {
-                    if (AirLoopInputsFilled) {
-                        UnitarySystems::UnitarySys::checkUnitarySysCoilInOASysExists(CompName, 0);
-                        MyOneTimeCheckUnitarySysFlag(OASysNum) = false;
-                    }
+                    UnitarySystems::UnitarySys::getUnitarySysHeatCoolCoil(CompName, OACoolingCoil, OAHeatingCoil, 0);
+                    UnitarySystems::UnitarySys::checkUnitarySysCoilInOASysExists(CompName, 0);
+                    if (Sim) MyOneTimeCheckUnitarySysFlag(OASysNum) = false;
```

Comment:
getInput had not completed at time of this call. And this call is useless after AirLoopInputsFilled is true. Also, this code is called twice (below) so hold off on setting MyOneTimeCheckUnitarySysFlag = false. Rearranged UnitarySystem getInput so that these calls would get the information they need.

    PrimaryAirSystem(AirSysNum).NumOAHeatCoils = GetOASysNumHeatingCoils(OANum);
    PrimaryAirSystem(AirSysNum).NumOACoolCoils = GetOASysNumCoolingCoils(OANum);

Reply:
Seems like some tricky sequencing; good find.